src/com/untamedears/realisticbiomes/GrowthConfig.java

```diff
@@ -320,17 +330,23 @@ public double getRate(Block block) {
 	
 	double sunlightChecks(Block block) {
 		double rate = 1.0;
-		int sunlightIntensity;
-		if (block.getType().isTransparent()) {
-			sunlightIntensity = block.getLightFromSky();
-		} else {
-			sunlightIntensity = block.getRelative(BlockFace.UP).getLightFromSky();
-		}
-		// apply multiplier if the sunlight is not at maximum
-		if (sunlightIntensity < MAX_LIGHT_INTENSITY) {
+		Block up = block.getRelative(BlockFace.UP);
+		int sunIntensity = up.getLightFromSky();
+		if(ignoreForLight == null) {
 			rate *= notFullSunlightMultiplier;
 			if (needsSunlight) {
-				rate *= Math.pow((sunlightIntensity / MAX_LIGHT_INTENSITY), 3.0);
+				rate *= Math.pow((up.getLightFromSky() / MAX_LIGHT_INTENSITY), 3.0);
+			}
+		} else {
+			while(sunIntensity < MAX_LIGHT_INTENSITY) {
+				if(!ignoreForLight.contains(block.getType())) {
+					rate *= notFullSunlightMultiplier;
+					if (needsSunlight) {
+						rate *= Math.pow((up.getLightFromSky() / MAX_LIGHT_INTENSITY), 3.0);
+					}
+					break;
+				}
+				up = up.getRelative(BlockFace.UP);
 			}
 		}

```

Comment:
1. If ignore_for_light is set, then `while(sunIntensity < MAX_LIGHT_INTENSITY)` is essentially a `while(true)` and should be changed to that, to avoid confusion.

2. `if(!ignoreForLight.contains(block.getType())) {` should use `up` instead of `block`. Right now it always checks the block (e.g. the sapling for being a leaf) so it never evaluates to true. The loop will exit after first iteration in all conditions (given that "leaf" is not a block that has a growth rate).

3. Assuming the `up` block is tested instead, it still will only have an effect if there is no air between the block and the leaf: the `while(true)` will exit as soon as there is something other than a block of `ignore_for_light`, including air. I believe the scenario is: sapling, some space, then leaves from a nearby tree?

Generally this part:

    rate *= notFullSunlightMultiplier;
    if (needsSunlight) {
      rate *= Math.pow((up.getLightFromSky() / MAX_LIGHT_INTENSITY), 3.0);
    }

Should not be duplicated and applied to either the block or the topmost leaf.